lib/internal/test_runner/test.js

```diff
@@ -409,17 +409,26 @@ class Test extends AsyncResource {
   }
 
   matchesTestNamePatterns() {
-    const matchesByNameOrParent = ArrayPrototypeSome(testNamePatterns, (re) =>
+    let patterns;
+    let boolOp = 0;
+    if (testNamePatterns !== null) {
+      patterns = testNamePatterns;
+    } else if (testSkipPatterns !== null) {
+      patterns = testSkipPatterns;
+      boolOp = 1;
+    } else return true; // Nothing is specified, just continue
+
+    const matchesByNameOrParent = ArrayPrototypeSome(patterns, (re) =>
       RegExpPrototypeExec(re, this.name) !== null,
     ) ||
-    this.parent?.matchesTestNamePatterns();
+      this.parent?.matchesTestNamePatterns();
 
-    if (matchesByNameOrParent) return true;
+    if (matchesByNameOrParent) return !!boolOp; // true
 
     const testNameWithAncestors = StringPrototypeTrim(this.getTestNameWithAncestors());
-    if (!testNameWithAncestors) return false;
+    if (!testNameWithAncestors) return !boolOp; // false
 
-    return ArrayPrototypeSome(testNamePatterns, (re) => RegExpPrototypeExec(re, testNameWithAncestors) !== null);
+    return ArrayPrototypeSome(patterns, (re) => RegExpPrototypeExec(re, testNameWithAncestors) !== null);
```

Comment:
Doesn't this branch also need to account for testSkipPatterns?

Reply:
`patterns` is set dynamically:
```
    let boolOp = 0;
    if (testNamePatterns !== null) {
      patterns = testNamePatterns;
    } else if (testSkipPatterns !== null) {
      patterns = testSkipPatterns;
      boolOp = 1;
    } else return true; // Nothing is specified, just continue
```

Reply:
Yes but isn't it the reverse operation? (That is checking there are no match if it's testSkipPatterns and if there is a match otherwise) like hte non-ancestor flow above?